### PR TITLE
Upgrade to CLDR 45

### PR DESCRIPTION
**Pull request by tomasr8**
[CLDR 45](https://cldr.unicode.org/index/downloads) is out now.
Looks like the 44 patch fixed most of the test issues so there was only one failing test this time around :)

Some related discussion here: #1070

### CODE REVIEW

**Review by akx — COMMENTED** [review]
LGTM, that was simple enough. 😄 

However, the [release notes](https://cldr.unicode.org/index/downloads/cldr-45) say "In Parent Locales, made substantial changes to the way that parentLocales work, including a new attribute for algorithmic handling of inheritance that avoids needing a long (and fragile) list of language-script codes to skip when falling back to root. That list was retained for migration, but will be withdrawn in the future."

Would you like to see whether that's something that affects us and implement whatever changes are relevant in `scripts/import_cldr.py`? If you don't have the bandwidth, it's fine, I can also follow up here.

**Comment by tomasr8** [comment]
> Would you like to see whether that's something that affects us and implement whatever changes are relevant in scripts/import_cldr.py? If you don't have the bandwidth, it's fine, I can also follow up here.

I might have some time this week, but feel free to have a look yourself if you want :)

**Comment by tomasr8** [comment]
I updated the PR to handle the parent locale changes. I just hope I understood it correctly :sweat_smile:

File: babel/localedata.py [hunk 1]
```diff
@@ -132,8 +153,11 @@ def load(name: os.PathLike[str] | str, merge_inherited: bool = True) -> dict[str
                 from babel.core import get_global
                 parent = get_global('parent_exceptions').get(name)
                 if not parent:
-                    parts = name.split('_')
-                    parent = "root" if len(parts) == 1 else "_".join(parts[:-1])
+                    if _is_non_likely_script(name):
+                        parent = 'root'
```
[the remark this answers is not in the document]
Comment (tomasr8) [reply babel/localedata.py:157]: This is already covered by the second test case with `az_Arab`. Do you want me to add another test for `_is_non_likely_script` as well?

[the remark this answers is not in the document]
Comment (tomasr8) [reply babel/localedata.py:157]: Any news on this @akx ? Do you think it's ok like this?

[the remark this answers is not in the document]
Comment (akx) [reply babel/localedata.py:157]: Sorry for the delay 😭 This is fine! :)

**Pull request merged**